src/components/molecules/PageFilter/PageFilter.tsx

```diff
@@ -77,26 +76,26 @@ const PageFilter = <T extends {}>({
 
   const [form] = Form.useForm(rest.form);
 
-  const parseSearch = useCallback(
+  const getSearch = useCallback(
     () =>
-      parseFilters<T>({
-        filters: search,
-        parseBoolean,
-        parseDates,
-        parseNumbers,
-      }),
-    [parseBoolean, parseDates, parseNumbers, search]
-  );
-
-  const [data, setData] = useState(
-    parseBoolean || parseDates || parseNumbers ? parseSearch() : search
+      parseDates || parseNumbers
+        ? parseFilters<T>({
+            filters: search,
+            parseDates,
+            parseNumbers,
+          })
+        : search,
+    [parseDates, parseNumbers, search]
   );
 
+  // Every time the search is updated
```

Comment:
maybe we should add this logic to a common Form wrapper around Antd Form and pass values to the form component that resets form based on new values ?

Reply:
i can see that we may use this logic in other places as well.

Reply:
Good idea, let's do that in a separate PR 👌